crates/nargo_cli/src/cli/test_cmd.rs

```diff
@@ -96,14 +96,17 @@ fn run_test<B: Backend>(
     show_output: bool,
     config: &CompileOptions,
 ) -> Result<(), CliError<B>> {
-    let mut program = compile_no_check(context, show_output, config, main)
-        .map_err(|_| CliError::Generic(format!("Test '{test_name}' failed to compile")))?;
+    let mut program = compile_no_check(context, config, main).map_err(|err| {
+        noirc_errors::reporter::report_all(&context.file_manager, &[err], config.deny_warnings);
```

Comment:
We were not reporting compile errors at all as `run_tests` assumes an error has already been printed. But we were only printing for execution failures.